src/types.rs

```diff
@@ -221,75 +160,73 @@ mod type_definition_tests {
+        IslType::named("my_int".to_owned(), vec![IslConstraint::type_constraint(IslTypeRef::anonymous(vec![IslConstraint::type_constraint(IslTypeRef::named("my_int".to_owned()))]))]),
+        TypeDefinition::named("my_int".to_owned(), vec![Constraint::type_constraint(1)])
     ),
     case::type_constraint_with_nested_type(
         /* For a schema with nested types as below:
             { name: my_int, type: { type: int } }
          */
-        IslType::NamedIslType(NamedIslType::new("my_int".to_owned(), vec![IslConstraint::Type(IslTypeRef::AnonymousType(AnonymousIslType::new(vec![IslConstraint::Type(IslTypeRef::CoreIslType(IonType::Integer))])))])),
-        TypeDefinition::NamedTypeDefinition(NamedTypeDefinition::new("my_int".to_owned(), vec![Constraint::Type(TypeConstraint::new(1))]))
+        IslType::named("my_int".to_owned(), vec![IslConstraint::type_constraint(IslTypeRef::anonymous(vec![IslConstraint::type_constraint(IslTypeRef::core_isl(IonType::Integer))]))]),
+        TypeDefinition::named("my_int".to_owned(), vec![Constraint::type_constraint(1)])
     ),
     case::type_constraint_with_nested_multiple_types(
         /* For a schema with nested multiple types as below:
             { name: my_int, type: { type: int }, type: { type: my_int } }
          */
-        IslType::NamedIslType(NamedIslType::new("my_int".to_owned(), vec![IslConstraint::Type(IslTypeRef::AnonymousType(AnonymousIslType::new(vec![IslConstraint::Type(IslTypeRef::CoreIslType(IonType::Integer))]))), IslConstraint::Type(IslTypeRef::AnonymousType(AnonymousIslType::new(vec![IslConstraint::Type(IslTypeRef::NamedType("my_int".to_owned()))])))])),
-        TypeDefinition::NamedTypeDefinition(NamedTypeDefinition::new("my_int".to_owned(), vec![Constraint::Type(TypeConstraint::new(1)), Constraint::Type(TypeConstraint::new(3))]))
+        IslType::named("my_int".to_owned(), vec![IslConstraint::type_constraint(IslTypeRef::anonymous(vec![IslConstraint::type_constraint(IslTypeRef::core_isl(IonType::Integer))])), IslConstraint::type_constraint(IslTypeRef::anonymous(vec![IslConstraint::type_constraint(IslTypeRef::named("my_int".to_owned()))]))]),
+        TypeDefinition::named("my_int".to_owned(), vec![Constraint::type_constraint(1), Constraint::type_constraint(3)])
     ),
     case::all_of_constraint(
         /* For a schema with all_of type as below:
             { all_of: [{ type: int }] }
         */
-        IslType::AnonymousIslType(AnonymousIslType::new(vec![IslConstraint::AllOf(vec![IslTypeRef::AnonymousType(AnonymousIslType::new(vec![IslConstraint::Type(IslTypeRef::CoreIslType(IonType::Integer))]))])])),
-        TypeDefinition::AnonymousTypeDefinition(AnonymousTypeDefinition::new(vec![Constraint::AllOf(AllOfConstraint::new(vec![1]))]))
+        IslType::anonymous(vec![IslConstraint::all_of(vec![IslTypeRef::anonymous(vec![IslConstraint::type_constraint(IslTypeRef::core_isl(IonType::Integer))])])]),
```

Comment:
Similar thought here RE: eliminating `to_owned()`s and `vec!`s. We can do this later, though.